examples/controlnet/README.md

```diff
@@ -0,0 +1,254 @@
+
+Using the pretrained models we can provide control images (for example, a depth map) to control Stable Diffusion text-to-image generation so that it follows the structure of the depth image and fills in the details. `train_controlnet.py` script shows how to train a controlnet from scratch with simple example data.
+
+This example is based on the [training example in the original Controlnet repository](https://github.com/lllyasviel/ControlNet/blob/main/docs/train.md) and it trains a controlnet that can fill circles using a small synthetic dataset.
+
+## Installing the dependencies
+
+Before running the scripts, make sure to install the library's training dependencies:
+
+**Important**
+
+To make sure you can successfully run the latest versions of the example scripts, we highly recommend **installing from source** and keeping the install up to date as we update the example scripts frequently and install some example-specific requirements. To do this, execute the following steps in a new virtual environment:
+```bash
+git clone https://github.com/huggingface/diffusers
+cd diffusers
+pip install -e .
+```
+
+Then cd in the example folder and run
+```bash
+pip install -r requirements.txt
+```
+
+And initialize an [🤗Accelerate](https://github.com/huggingface/accelerate/) environment with:
+
+```bash
+accelerate config
+```
+
+Or for a default accelerate configuration without answering questions about your environment
+
+```bash
+accelerate config default
+```
+
+Or if your environment doesn't support an interactive shell e.g. a notebook
+
+```python
+from accelerate.utils import write_basic_config
+write_basic_config()
+```
+
+## Circle filling dataset
+
+The original dataset is hosted in the [controlnet repo](https://huggingface.co/lllyasviel/ControlNet/blob/main/training/fill50k.zip). We re-uploaded it to be compatible with datasets [here](https://huggingface.co/datasets/fusing/fill50k). Note that datasets handles dataloading
+within the training script.
+
+Our training examples use stable diffusion 1.5 as the original set of controlnet models were trained from it. However, controlnet can be trained to augment any stable diffusion compatible model.
+
+## Training
+
+```bash
+export MODEL_DIR="runwayml/stable-diffusion-v1-5"
+export OUTPUT_DIR="path to save model"
+
+accelerate launch train_controlnet.py \
+ --pretrained_model_name_or_path=$MODEL_DIR \
+ --output_dir=$OUTPUT_DIR \
+ --dataset_name=fusing/fill50k \
+ --resolution=512 \
+ --learning_rate=1e-5 \
+ --validation_image "./images/conditioning_image_1.png" "./images/conditioning_image_2.png" \
+ --validation_prompt "red circle with blue background" "cyan circle with brown floral background" \
+ --train_batch_size=4
+```
+
+This default configuration requires ~38GB VRAM.
+
+Gradient accumulation with a smaller batch size can be used to reduce training requirements to ~20 GB VRAM
```

Comment:
```suggestion
Gradient accumulation with a smaller batch size can be used to reduce training requirements to ~20 GB VRAM.
```

What happens when using `xformers`? Maybe we should just add a note like:

> We haven't rigorously test all the optimization techniques supported by `diffusers` for training but we think it should be possible to further reduce the memory footprint. This can be done by using mixed-precision, memory-efficient attention with `xformers`, and, 8-bit Adam for training.

WDYT?